### ISSUE TITLE
tickRound: false should also disable rounding on gridlines

### ISSUE THREAD

**Issue by donghaoren** [vega/vega#2355]
~`tickRound` is supported, but the corresponding `gridRound` is missing. If we set `tickRound: true` and have grid lines, the ticks and grid lines doesn't line up.~

When tickRound is set to false, tick positions are not rounded but grid lines are still rounded. This result in misalignment between ticks and gridlines.

![image](https://user-images.githubusercontent.com/1595237/73588287-047a9f80-447c-11ea-9515-492ab9f2aee6.png)

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "width": 600,
  "height": 200,
  "data": {"url": "data/unemployment-across-industries.json"},
  "mark": "bar",
  "encoding": {
    "x": {
      "timeUnit": "yearquarter",
      "field": "date",
      "band": 0.6,
      "type": "temporal",
      "axis": {"format": "%Y", "tickRound": false}
    },
    "y": {
      "aggregate": "sum",
      "field": "count",
      "type": "quantitative",
      "axis": {"title": "count"}
    }
  }
}
```

In this I'd like the tick & grid line to appear exactly at the middle between bars, but right now it looks like this:
![image](https://user-images.githubusercontent.com/1595237/73588296-2116d780-447c-11ea-972a-2b28120067c8.png)

**Comment by donghaoren** [comment]
Updated title and description.

**Comment by jheer** [comment]
I don't think this is a Vega bug. If you look at vega-parser/src/parsers/guides/axis-grid.js, you'll see that we already pass tickRound through to the grid lines. Instead, it looks like the tickRound parameter is being applied to the ticks layer axis but not the grid line layer axis in Vega-Lite.

**Comment by domoritz** [comment]
@donghaoren is this still an issue? If so, we should file/fix in Vega-Lite.

**Comment by domoritz** [comment]
Since this is a Vega-Lite issue, I'll reopen and transfer.
